### PR TITLE
1930 Add new microbenchmarks

### LINKED ISSUE #1930: Add new microbenchmarks
Add microbenchmarks for performance analysis.

### PR DESCRIPTION
**Pull request by stmcgovern**
Closes #1930. Depends on changes to runnable (e.g., makeRunnable, RunnableNew) from #1899 and #1914, and possibly others.

### CODE REVIEW

File: tests/perf/make_runnable_micro.cc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    perfRunBenchmark();
+    test_obj_->StopTimer(fmt::format("makeRunnable {}", num_iters));
+
+    theTerm()->any_epoch_state_.decrementDependency();
+  }
+
+  void perfRunBenchmark() {
+    for (int i = 0; i < num_iters; i++) {
+      {
+	bool add_context = true;
```
Comment (lifflander) [inline tests/perf/make_runnable_micro.cc:93]: We should remove this `add_context` completely. It was just for some incremental performance testing.

File: tests/perf/make_runnable_micro.cc [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+private:
+  std::vector<MsgSharedPtr<MyMsg>> msgs;
+  HandlerType han;
+  MyTest* test_obj_ = nullptr;
+  vt::objgroup::proxy::Proxy<NodeObj> proxy_ = {};
+  int reduce_counter_ = -1;
+  int i = 0;
+};
+
+VT_PERF_TEST(MyTest, test_ping_pong) {
```
Comment (lifflander) [inline tests/perf/make_runnable_micro.cc:113]:
```suggestion
VT_PERF_TEST(MyTest, test_make_runnable) {
```

File: tests/perf/make_runnable_micro.cc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  }
+  
+  void perfMakeRunnable(MyMsg* in_msg) {
+    for (int i = 0; i < num_iters; i++) {
+      msgs.emplace_back(makeMessage<MyMsg>());
+    }
+
+    han = auto_registry::makeAutoHandler<MyMsg, &dummyHandler>();
+
+    theTerm()->any_epoch_state_.incrementDependency();
```
Comment (lifflander) [inline tests/perf/make_runnable_micro.cc:81]: I want to refactor this into a difference PR to disable the TD (this accesses private state, which was manually modified).

File: tests/perf/make_runnable_micro.cc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+  void perfRunBenchmark() {
+    for (int i = 0; i < num_iters; i++) {
+      {
+	bool add_context = true;
+	auto r = runnable::makeRunnable(msgs[i], false, han, 0, add_context)
+	  .withContinuation(nullptr)
+	  .withTag(TagType{0})
+	  .withTDEpochFromMsg(false);
+	r.enqueue();
```
Comment (lifflander) [inline tests/perf/make_runnable_micro.cc:98]: I would like to somehow parameterize this over just `.run` vs. `.enqueue` and then the runScheduler.

**Review by lifflander — APPROVED** [review]

**Review by cz4rs — APPROVED** [review]
Looks good to me.

**Comment by stmcgovern** [comment]
I split the test collection_local_send into two files (for handling message creation differently), since the test_harness is designed to just have one test per file. I don't think the statistics would work for 2 test configuration cases, as well.

**Review by lifflander — APPROVED** [review]

**Pull request merged**